dateparser/timezones.py

```diff
@@ -6,7 +6,7 @@
 timezone_info_list = [
     {
         'regex_patterns':
-            [r'(\W|\d|_)\(?%s\)?$'],
+            [r'(\W|\d|_)\(?%s(\)|$|[^\w+])'],
```

Comment:
Okay, so timezones is *internal* mechanics of our parser working with `dateutil` and applied only there. I see...

I really don't like existing solution of just adding `(` to regex because of `17:00 (EDT)`, `17:00 [EDT]`, `17:00 {EDT}`, `17:00 <EDT>`, `17:00 | EDT`, `17:00 ^EDT` and so on... Any ideas?